tools/gulp/watch.js

```diff
@@ -30,7 +30,7 @@ module.exports = (gulp, shared) => {
       `packages/${packages}/src/**/*.{js,jsx}`,
       `!packages/${packages}/src/**/*.example.{js,jsx}`,
       `!packages/${packages}/src/**/*.test.{js,jsx}`
-    ], ['lint:packages-scripts', 'docs:react']);
+    ], ['lint:packages-scripts', 'sass:process:docs', 'docs:generate-pages', 'docs:react']);
```

Comment:
This is probably a little confusing based on how things are currently named, but `docs:generate-pages` will need to come after `docs:react`, since `docs:react` creates a file (`react-doc.json`) which the `docs:generate-pages` task depends upon.

Also, I don't think `sass:process:docs` needs called here or below. It only transforms the Sass to CSS, which wouldn't need done when a JSX file changes.

Reply:
Ah, I assumed `sass:process:docs` read the docs out of the sass